test/test_prerun.py

```diff
@@ -11,35 +13,113 @@
 from ansiblelint.constants import INVALID_PREREQUISITES_RC
 from ansiblelint.testing import run_ansible_lint
 
+EXECDIR = os.path.dirname(sys.executable)
+
+Deactivator = Callable[[bool], str]
+DeactivateVenv = Callable[[], Deactivator]
+
+
+@pytest.fixture
+def deactivate_venv(monkeypatch: MonkeyPatch) -> Deactivator:
+    """Deliver a function to deactivate the current venv, if any, if requested, by removing it's bin dir from $PATH."""
+
+    def deactiveator(deactivate: bool) -> str:
+        if deactivate:
```

Comment:
Pro tip: this clause can be turned into a "guard expression" by flipping the check. Use  `if not request.param: return ""` to exit early and this will allow dedenting a whole bunch of code making the control flow look more linear.